builtin/credential/aws/path_login.go

```diff
@@ -37,22 +46,50 @@ If a matching role is not found, login fails.`,
 			},
 
 			"pkcs7": {
-				Type:        framework.TypeString,
-				Description: "PKCS7 signature of the identity document.",
+				Type: framework.TypeString,
+				Description: `PKCS7 signature of the identity document when using an auth_type
+of ec2.`,
 			},
 
 			"nonce": {
 				Type: framework.TypeString,
-				Description: `The nonce to be used for subsequent login requests.
-If this parameter is not specified at all and if reauthentication is allowed,
-then the backend will generate a random nonce, attaches it to the instance's
-identity-whitelist entry and returns the nonce back as part of auth metadata.
-This value should be used with further login requests, to establish client
-authenticity. Clients can choose to set a custom nonce if preferred, in which
-case, it is recommended that clients provide a strong nonce.  If a nonce is
-provided but with an empty value, it indicates intent to disable
-reauthentication. Note that, when 'disallow_reauthentication' option is enabled
-on either the role or the role tag, the 'nonce' holds no significance.`,
+				Description: `The nonce to be used for subsequent login requests when
+auth_type is ec2.  If this parameter is not specified at
+all and if reauthentication is allowed, then the backend will generate a random
+nonce, attaches it to the instance's identity-whitelist entry and returns the
+nonce back as part of auth metadata.  This value should be used with further
+login requests, to establish client authenticity. Clients can choose to set a
+custom nonce if preferred, in which case, it is recommended that clients provide
+a strong nonce.  If a nonce is provided but with an empty value, it indicates
+intent to disable reauthentication. Note that, when 'disallow_reauthentication'
+option is enabled on either the role or the role tag, the 'nonce' holds no
+significance.`,
+			},
+
+			"iam_http_request_method": {
+				Type: framework.TypeString,
+				Description: `HTTP method to use for the AWS request when auth_type is
+iam. This must match what has been signed in the
+presigned request. Currently, POST is the only supported value`,
+			},
+
+			"iam_request_url": {
+				Type: framework.TypeString,
+				Description: `Base64-encoded full URL against which to make the AWS request
+when using iam auth_type. If using a POST request with the action specified in the
+body, this should just be "Lw==" ("/" base64-encoded).`,
```

Comment:
Rather than require this special value if using a POST request, can we just check for the value being empty or being this and do the right thing in either case? Or maybe just add a `Default` field with that value?